datajunction-clients/python/datajunction/builder.py

```diff
@@ -76,6 +84,63 @@ def restore_namespace(self, namespace: str, cascade: bool = False) -> None:
     #
     # Nodes: all
     #
+    def create_node(
+        self,
+        type_: models.NodeType,
+        name: str,
+        data: Dict,
+        update_if_exists: bool = False,
+    ) -> "Node":
+        """
+        Create or update a new node
+        """
+
+        node_type_to_class: dict[models.NodeType, Type[Node]] = {
+            models.NodeType.METRIC: Metric,
+            models.NodeType.CUBE: Cube,
+            models.NodeType.TRANSFORM: Transform,
+            models.NodeType.SOURCE: Source,
+            models.NodeType.DIMENSION: Dimension,
+        }
+
+        data = {
+            k: v
+            for k, v in data.items()
+            if k not in ["dj_client", "type"] and v is not None
+        }
+
+        try:
+            existing_node_dict = self._get_node(name)
+        except DJClientException:  # pragma: no cover # pytest fixture doesn't raise
```

Comment:
Do you want to check here if the `DJClientException` actually was a "node does not exist" error?

Reply:
+1

Reply:
Fixed. Since there's no specific exception for "NodeAlreadyExists", I inspect the DJClientException message with regex.